### PR TITLE
Added guard, changed routing style.

### PR DESCRIPTION
**Pull request by blrubenstein**

### CODE REVIEW

File: ui/src/app/guards/sign-in-guard.service.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import {SignInService} from 'app/services/sign-in.service';
+
+declare const gapi: any;
+
+@Injectable()
+export class SignInGuard implements CanActivate, CanActivateChild {
+  constructor(private signInService: SignInService, private router: Router) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+    return new Observable((observer) => {
```
Comment (johnstill) [inline ui/src/app/guards/sign-in-guard.service.ts]: I think this is equivalent to just `return this.signInService.user.pluck('isSignedIn');`; I'm not sure you need the wrapping / unwrapping with another Observable.  If it complains about type info, you can try `pluck<any, boolean>()`

File: ui/src/app/guards/sign-in-guard.service.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      this.signInService.user.subscribe(user => {
+
```
Comment (johnstill) [inline ui/src/app/guards/sign-in-guard.service.ts]: I assume that for the purposes of chunking the work this PR is intentionally ignoring anything to do with redirecting, correct?

Comment (blrubenstein) [reply ui/src/app/guards/sign-in-guard.service.ts]: Correct. About to leave the house, but I’ll get to other comments when I get back.

File: ui/src/app/guards/sign-in-guard.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  CanActivateChild,
+  Router, RouterStateSnapshot
+} from '@angular/router';
+
+import {Observable} from 'rxjs/Observable';
+
+
+import {SignInService} from 'app/services/sign-in.service';
+
+declare const gapi: any;
```
Comment (johnstill) [inline ui/src/app/guards/sign-in-guard.service.ts]: Btw is this declaration needed b/c TypeScript wants it here or is it left over from something else?

Comment (blrubenstein) [reply ui/src/app/guards/sign-in-guard.service.ts]: I believe its needed by typescript/the compiler. It didn't know what gapi was without that.

**Pull request merged**